arangod/Cluster/ClusterInfo.cpp

```diff
@@ -1970,9 +1968,15 @@ void ClusterInfo::loadCurrent() {
 
       for (auto const& shardSlice :
            velocypack::ObjectIterator(collectionSlice.value)) {
-        pmr::ManagedString shardID{shardSlice.key.copyString(),
-                                   _resourceMonitor};
 
+        auto maybeShardID = ShardID::shardIdFromString(shardSlice.key.copyString());
+        if (ADB_UNLIKELY(maybeShardID.fail())) {
+          TRI_ASSERT(false) << "Indexed malformed shard name "<< shardSlice.key.copyString();
+          // TODO cannot handle this entry, is it better to continue or to abort here?
+          continue;
+        }
+
+        auto shardID = std::move(maybeShardID.get());
```

Comment:
`maybeShardID.get()` is already an rvalue. No need to add an extra `std::move()`.
```suggestion
        auto shardID = maybeShardID.get();
```